src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -88,12 +90,33 @@ public static async Task<GeolocationAccessStatus> RequestAccessAsync()
 				else
 				{
 					BroadcastStatus(PositionStatus.Disabled);
+
+					foreach (var subscriber in _positionChangedSubscriptions)
+					{
+						subscriber.Key.WaitForPermissionFromBackground();
+					}
 				}
 			}
 
 			return status;
 		}
 
+		public void WaitForPermissionFromBackground()
+		{
+			CoreApplication.Resuming -= CoreApplication_Resuming;
+			CoreApplication.Resuming += CoreApplication_Resuming;
+		}
+
+		private async void CoreApplication_Resuming(object sender, object e)
```

Comment:
Please schedule in the dispatcher instead, this may crash this application (async void)